### DISCUSSION TITLE
Lesson 4 - Price Feed Sample Contract Error

### DISCUSSION THREAD

**Discussion by RichardPalladino** [Cyfrin/foundry-full-course-cu#139] [Q&A] [answered]
I'm trying to follow along with the lesson (which I know is dated at this point) and get the PriceConsumer.sol sample contract from ChainLink running on Sepolia.  

- I opened the sample in Remix.  (Current sample says it's for mainnet.)
- I found the feed registry contract address for ETH/USD on Sepolia
- I deploy successfully without any error but `getEthusdPrice()` always fails with _{"jsonrpc":"2.0","error":"execution reverted","id":1317729414942697_  (or some unique ID number)

Has anyone tried to get the sample to work on Sepolia?  Anyone else found this problem and solved it?

Thanks.

**Comment by alymurtazamemon** [comment]
@RichardPalladino Which sample contract are you talking about? Can you point us to that? The one in the code example does not have any `getEthusdPrice` function in chainlink docs.

**Reply by RichardPalladino** [reply]
I was specifically referring to Chainlink's own example code, which the video referenced, located on their site:
["Feed Registry - Code Examples"](https://docs.chain.link/data-feeds/feed-registry)

I have copied the sample code below, and I deploy it with the Sepolia feed registry described in the updated comments:

```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.7;

import "@chainlink/contracts/src/v0.8/interfaces/FeedRegistryInterface.sol";
import "@chainlink/contracts/src/v0.8/Denominations.sol";

/**
 * THIS IS AN EXAMPLE CONTRACT THAT USES HARDCODED VALUES FOR CLARITY.
 * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
 * DO NOT USE THIS CODE IN PRODUCTION.
 */

contract PriceConsumer {
    FeedRegistryInterface internal registry;

    /**
     * Network: Sepolia 
     * Feed Registry: 0x694AA1769357215DE4FAC081bf1f309aDC325306
     */
    constructor(address _registry) {
        registry = FeedRegistryInterface(_registry);
    }

    /**
     * Returns the ETH / USD price
     */
    function getEthUsdPrice() public view returns (int) {
        // prettier-ignore
        (
            /*uint80 roundID*/,
            int price,
            /*uint startedAt*/,
            /*uint timeStamp*/,
            /*uint80 answeredInRound*/
        ) = registry.latestRoundData(Denominations.ETH, Denominations.USD);
        return price;
    }

    /**
     * Returns the latest price
     */
    function getPrice(address base, address quote) public view returns (int) {
        // prettier-ignore
        (
            /*uint80 roundID*/,
            int price,
            /*uint startedAt*/,
            /*uint timeStamp*/,
            /*uint80 answeredInRound*/
        ) = registry.latestRoundData(base, quote);
        return price;
    }
}
```

**Comment by ShayanShamsi** [comment, marked as answer]
Hey @RichardPalladino. I think you have confused what was being used in the video with Feed Registry which is a different Chainlink service and is available only on Ethereum Mainnet currently as indicated [here](https://docs.chain.link/data-feeds/feed-registry#contract-addresses). To my understanding, the key element which distinguishes Feed Registry is the fact that you can get the latest price of an asset from just **one contract** rather than having to use multiple contracts each for a different asset.

To follow what was being done in the video, you should use Data Feeds that are explained [here](https://docs.chain.link/data-feeds/using-data-feeds#overview). You can see [here](https://docs.chain.link/data-feeds/price-feeds/addresses) that these contracts are available on multiple chains (unlike Feed Registry which is only on Ethereum mainnet).

**Reply by RichardPalladino** [reply]
Yup.  That's the correct answer.  Thanks Shayan.
